memory/test/test_memory.py

```diff
@@ -0,0 +1,58 @@
+import concurrent
+import os
+import unittest
+import uuid
+from memory.client import MemoryClient
+
+from hailtop.config import get_user_config
+from hailtop.google_storage import GCS
+from hailtop.utils import async_to_blocking
+
+
+class BlockingMemoryClient:
+    def __init__(self, gcs_project=None, fs=None, deploy_config=None, session=None,
+                 headers=None, _token=None):
+        self._client = async_to_blocking(MemoryClient(gcs_project, fs, deploy_config, session, headers, _token))
+
+    def _get_file_if_exists(self, filename):
+        return async_to_blocking(self._client._get_file_if_exists(filename))
+
+    def read_file(self, filename):
+        return async_to_blocking(self._client.read_file(filename))
+
+    def close(self):
+        return async_to_blocking(self._client.close())
+
+class Tests(unittest.TestCase):
```

Comment:
Although I have mixed feelings about it, I'm a pytest convert.  It also makes async tests very easy, and has better support for fixtures (setup and teardown).  You don't need to change it in this PR, but we should switch afterwards.  It will also remove the need for a blocking client.  See my recent FS per for examples:

https://github.com/hail-is/hail/blob/208fbe2c20682590779d67dbf1c4dbf077cee0f2/hail/python/test/hailtop/test_aiogoogle.py